sklearn/manifold/t_sne.py

```diff
@@ -403,17 +625,42 @@ def __init__(self, n_components=2, perplexity=30.0,
         self.init = init
         self.verbose = verbose
         self.random_state = random_state
+        self.method = method
+        self.angle = angle
+        self.embedding_ = None
 
-    def _fit(self, X):
+    def _fit(self, X, skip_num_points=0):
         """Fit the model using X as training data.
 
         Parameters
         ----------
         X : array, shape (n_samples, n_features) or (n_samples, n_samples)
             If the metric is 'precomputed' X must be a square distance
-            matrix. Otherwise it contains a sample per row.
+            matrix. Otherwise it contains a sample per row. Note that this
+            when method='barnes_hut', X cannot be a sparse array and if need be
+            will be converted to a 32 bit float array. Method='standard' allows
+            sparse arrays and 64bit floating point inputs.
+
+        skip_num_points : int (optional, default:0)
+            This does not compute the gradient for points with indices below
+            `skip_num_points`. This is useful when computing transforms of new
+            data where you'd like to keep the old data fixed.
         """
-        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
+        if self.method not in ['barnes_hut', 'standard']:
+            raise ValueError("'method' must be 'barnes_hut' or 'standard'")
+        if self.angle < 0.0 or self.angle > 1.0:
+            raise ValueError("'angle' must be between 0.0 - 1.0")
+        if self.method == 'barnes_hut' and self.n_components > 3:
+            raise ValueError("method='barnes_hut' only available for "
+                             "n_components < 4.")
+        if self.method == 'barnes_hut' and sp.issparse(X):
+            raise TypeError('A sparse matrix was passed, but dense '
+                            'data is required. Use X.toarray() to '
+                            'convert to a dense numpy array or '
```

Comment:
Maybe it would be smarter to recommend to transform the sparse data into a 50 to 100 dimensional sparse using the `TruncatedSVD` transformer for instance.